camel-openmrs/src/main/java/org/openmrs/eip/component/camel/OpenmrsAttachmentProducer.java

```diff
@@ -0,0 +1,33 @@
+package org.openmrs.eip.component.camel;
+
+import java.io.File;
+
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.attachment.AttachmentMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+
+public class OpenmrsAttachmentProducer extends AbstractOpenmrsProducer {
+	private static Logger log = LoggerFactory.getLogger(OpenmrsAttachmentProducer.class);
+	
+	public OpenmrsAttachmentProducer(final OpenmrsEndpoint endpoint, final ApplicationContext applicationContext, final ProducerParams params) {
+		super(endpoint, applicationContext, params);
+	}
+	
+	@Override
+	public void process(final Exchange exchange) {
+		log.info("Attaching the log file to email");
+		
+		AttachmentMessage in = exchange.getMessage(AttachmentMessage.class); 
```

Comment:
Do you really want to share the log file via email? The file might also be too big, I think the exception type that was thrown and stacktrace are enough

Reply:
Sometime you need the whole log to understand the context of the Exception. So to avoid to go back to site and ask for the log could be painful. The log file is setup to limit of 10mb. If this is too much we can compress it.

Reply:
OK, that makes sense especially from the context of the remote sites, and yes I guess you are right the log file size is limited